sui-execution/latest/sui-adapter/src/temporary_store.rs

```diff
@@ -1015,13 +1024,21 @@ impl<'backing> Storage for TemporaryStore<'backing> {
     fn check_coin_deny_list(
         &self,
         written_objects: &BTreeMap<ObjectID, Object>,
-    ) -> Result<(), ExecutionError> {
-        // TODO: How do we track runtime loaded objects for replay?
-        check_coin_deny_list_v2_during_execution(
+    ) -> (Result<(), ExecutionError>, u64) {
+        let (result, num_regulated_transfers) = check_coin_deny_list_v2_during_execution(
             written_objects,
             self.cur_epoch,
             self.store.as_object_store(),
-        )
+        );
+        // The denylist object is only loaded if there are regulated transfers.
+        // And also if we already have it in the input there is no need to commit it again in the effects.
+        if num_regulated_transfers > 0 && !self.input_objects.contains_key(&SUI_DENY_LIST_OBJECT_ID)
+        {
+            self.loaded_per_epoch_config_objects
+                .write()
+                .insert(SUI_DENY_LIST_OBJECT_ID);
```

Comment:
We do not have any of these APIs exposed yet, and it is a problem for that time, but we will need something more general here for configs, yeah? Since we will allow reading directly from the config (unsequenced) within Move. 
Given that, should this insert the config ID for each regulated coin type? Or are we fine with just the deny list object for this since it is a special case?

Reply:
Good question. I probably don't need to do this anymore because the indexer team could special handle 0x403 anyway. Down the road I think we need to be able to rely on the data in effects for the query

Reply:
Oh oh I confused myself. We need this here. And in the future, for any root config object ID we will need to add it here to this data structure, so that they get committed to the effects

Reply:
Fine with as is, my question was more should we be putting the individual configs here, rather than the deny list. But just putting the denylist seems simpler

Reply:
I don't think we can put the individual configs here, because a newly created config might be found (but not have a current-epoch value) on some validators, and not found at all on others.

Reply:
Yeah what Mark said, we cannot put child objects here.